llvm/unittests/SandboxIR/SandboxIRTest.cpp

```diff
@@ -2037,3 +2037,32 @@ define void @foo(i32 %arg) {
   }
   EXPECT_EQ(NewPHI->getNumIncomingValues(), PHI->getNumIncomingValues());
 }
+
+TEST_F(SandboxIRTest, UnreachableInst) {
+  parseIR(C, R"IR(
+define void @foo() {
+  ret void
+  unreachable
+}
+)IR");
+  llvm::Function *LLVMF = &*M->getFunction("foo");
+  sandboxir::Context Ctx(C);
+  sandboxir::Function *F = Ctx.createFunction(LLVMF);
+  auto *BB = &*F->begin();
+  auto It = BB->begin();
+  auto *Ret = &*It++;
+  auto *UI = cast<sandboxir::UnreachableInst>(&*It++);
+
+  EXPECT_EQ(UI->getNumSuccessors(), 0u);
+  EXPECT_EQ(UI->getNumOfIRInstrs(), 1u);
+  // Check create(InsertBefore)
+  sandboxir::UnreachableInst *NewUI =
+      sandboxir::UnreachableInst::create(/*InsertBefore=*/Ret, Ctx);
+  EXPECT_NE(NewUI, nullptr);
+  EXPECT_EQ(NewUI->getParent(), BB);
+  // Check create(InsertAtEnd)
+  sandboxir::UnreachableInst *NewUIEnd =
```

Comment:
maybe we shouldn't do a test for InsertAtEnd and remove the `static UnreachableInst *create(BasicBlock *InsertAtEnd, Context &Ctx);` function? 

the bot is failing due to 
```
SandboxIR.cpp:289: BBIterator &llvm::sandboxir::BBIterator::operator++(): Assertion `It != ItE && "Already at end!"' failed.
```

Reply:
It might have to do with the fact that in the ll code above you have placed `unreachalbe` after `ret`.
Could you try the opposite:
```
define void @foo() {
  unreachable
  ret void
}
```

Reply:
presuming I have to also change 
```
  auto *UI = cast<sandboxir::UnreachableInst>(&*It++);
  auto *Ret = &*It++;
```
to
```
  auto *Ret = &*It++;
  auto *UI = cast<sandboxir::UnreachableInst>(&*It++);
```
I encountered a seg fault. So I decided to add another instruction to help alleviate 
```Assertion `It != ItE && "Already at end!"' failed.```

now my LL looks like this

```
define void @foo() {
  call void @llvm.donothing()
  unreachable
  ret void
}
```

Reply:
Thanks for the heads up, it seems like it's being accepted by the bot.

Reply:
> presuming I have to also change

Yes, change it to:
```
  auto *UI = cast<sandboxir::UnreachableInst>(&*It++);
  auto *Ret = `cast<sandboxir::ReturnInst>(&*It++);
```

You shouldn't need another instruction. Calling create with `/*InsertBefore=*/Ret` should work fine.

What kind of crash are you getting with this code? Could you share the backtrace?